src/Plugin/WSConnector/WSConnectorSimpleHTTP.php

```diff
@@ -73,21 +95,78 @@ public function getReplacements(array $options) {
   /**
    * {@inheritdoc}
    */
-  public function getOptionsForm() {
+  public function getOptionsForm($options = []) {
+
     $methods = $this->getMethods();
+    $form['path'] = [
+      '#title' => t('Path'),
+      '#description' => t('The final endpoint will be <em>Server Endpoint/Path</em>'),
+      '#type' => 'textfield',
+      '#maxlength' => 512,
+    ];
 
-    return [
-      'path' => [
-        '#title' => t('Path'),
-        '#description' => t('The final endpoint will be <em>Server Endpoint/Path</em>'),
-        '#type' => 'textfield',
-      ],
-      'method' => [
-        '#title' => t('HTTP Method'),
-        '#type' => 'select',
-        '#options' => array_combine($methods, $methods),
-      ],
+    $form['method'] = [
+      '#title' => t('HTTP Method'),
+      '#type' => 'select',
+      '#options' => array_combine($methods, $methods),
     ];
+
+    $header_count = 1;
+
+    if (isset($options['form_state'])) {
+      if ($options['form_state']->getUserInput()['headers_count']) {
+        $header_count = $options['form_state']->getUserInput()['headers_count'] + 1;
+      }
+    }
+
+    $form['body'] = [
```

Comment:
Body shouldn't be here.  Body is the payload.  WSdata->call has a special argument just for it.  Also, how would the encoder/decoder get applied?

Reply:
You are right, I am going to move this into the custom form that way the payload can be modified for each block/field.